Push/PushDecrypt.swift

```diff
@@ -0,0 +1,94 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import Foundation
+import FxA
+
+/// Class to wrap ece.c which does the decryption with OpenSSL.
+/// This supports aes128gcm and aesgcm.
+/// This will also support the generation of keys to register with a push server.
+class PushDecrypt {
+    func aes128gcm(payload data: String, decryptWith privateKey: String, authenticateWith authKey: String) throws -> String {
+        var authSecret = try stringToBuffer(authKey)
+        var rawRecvPrivKey = try stringToBuffer(privateKey)
+        var plaintext = emptyBuffer()
+        var payload = try stringToBuffer(data)
+
+        defer {
+            ece_buf_free(&authSecret)
+            ece_buf_free(&rawRecvPrivKey)
+            ece_buf_free(&payload)
+            ece_buf_free(&plaintext)
+        }
+
+        let err =
+            ece_aes128gcm_decrypt(&rawRecvPrivKey, &authSecret, &payload, &plaintext)
+        if (err != 0) {
+            throw PushDecryptError.cantDecrypt
+        }
+        
+        return try bufferToString(plaintext)
+    }
+
+    func aesgcm(ciphertext data: String, decryptWith privateKey: String, authenticateWith authKey: String, encryptionHeader: String, cryptoKeyHeader: String) throws -> String {
+
+        var authSecret = try stringToBuffer(authKey)
+        var rawRecvPrivKey = try stringToBuffer(privateKey)
+        var plaintext = emptyBuffer()
+        var ciphertext = try stringToBuffer(data)
+
+        defer {
+            ece_buf_free(&authSecret)
+            ece_buf_free(&rawRecvPrivKey)
+            ece_buf_free(&ciphertext)
+            ece_buf_free(&plaintext)
+        }
+
+        let err = ece_aesgcm_decrypt(&rawRecvPrivKey, &authSecret, cryptoKeyHeader,
+                           encryptionHeader, &ciphertext, &plaintext)
+        if (err != 0) {
+            throw PushDecryptError.cantDecrypt
+        }
+
+        return try bufferToString(plaintext)
+    }
+}
+
+/// Some utility methods that make package up dealing with the C API a little easier.
+extension PushDecrypt {
```

Comment:
Nit: If these helper methods aren't method to be exposed you could make this extension private.